### ISSUE TITLE
Include testing for packages containing native DLLs > 2GB

### ISSUE THREAD

**Issue by dsyme** [dotnet/sdk#18507]
Recommendation: Add end-to-end tests for restoring and compiling against packages that have native DLLs > 2GB

Background:

.NET 5.0 SDK had [a bug](https://github.com/dotnet/runtime/pull/50237#issuecomment-866846085) where it can't cope with packages that have native DLLs > 2GB.  This is fixed in .NET 6.0 however I'd imagine there's no end-to-end testing covering such a case.

You might ask, how can there possibly be a native DLL > 2GB?  Well, such monstrosities are now happening in AI Runtimes such as PyTorch which includes the absolutely vast native binary `libtorch\lib\libtorch_cuda_cpp.so`, which in recent releases is now > 2GB and growing about 30% every release. Evidently it contains pre-compiled kernels for every imaginable tensor kind, shape, size, operation, whatever. Quite remarkable but it is what it is, the great new world of AI is upon us

Such packages are rare because they can't yet be hosted on nuget.org, however the TorchSharp repo uses various hacks to split the monster component into multiple packages and restitch them on restore. Hence we do encounter such a large beast even when using nuget.org packages.

Anyway I'd like to recommend we have proper testing in place for dealing with nuget packages that include native dependencies > 2GB in size. I fear we may have trouble in the future if this is not under test

cc @rainersigwald

**Comment by wli3** [comment]
I think the test would be pretty slow. Especially we are running tests in parallel in the cloud. Considering we don't have too many bugs in this area (or users in this area), I prefer postpone this task until the code is proven fragile, or there are some solution can reduce the cost.

**Comment by dsyme** [comment]
Given how bad a bug is in this scenario, and how likely it is to regress given the range of failures possible, I really think this is something the .NET SDK should be testing end to end.  The product is meant to succeed in this scenario, and in which case it should be under test.

Note a 3GB package should not take long in a data-centre.  We restore them all the time in Azure automated tests.

**Comment by dsyme** [comment]
If you like we can simply automate the restore of these, with an appropriate small C# code to compile against them (that step is important).  I can provide the test if you like and we can see how long it takes to run - I think it will be ok in the data centres

* https://www.nuget.org/packages/libtorch-cuda-11.1-win-x64/1.9.0.10
* https://www.nuget.org/packages/TorchSharp/0.91.52719


also 
* https://www.nuget.org/packages/libtorch-cuda-11.1-linux-x64/1.9.0.10
* https://www.nuget.org/packages/TorchSharp/0.91.52719

Thanks

**Comment by marcpopMSFT** [comment]
Restore is already the least reliable part of our testing infrastructure and I'd be wary of adding more to this.  We could potentially add this to the manual testing process @richaverma1
